stdlib/posixpath.pyi

```diff
@@ -60,25 +61,29 @@ __all__ = [
 
 supports_unicode_filenames: bool
 # aliases (also in os)
-curdir: str
-pardir: str
-sep: str
-altsep: str | None
-extsep: str
-pathsep: str
-defpath: str
-devnull: str
+curdir: LiteralString
+pardir: LiteralString
+sep: LiteralString
+altsep: LiteralString | None
+extsep: LiteralString
+pathsep: LiteralString
+defpath: LiteralString
+devnull: LiteralString
 
 # Overloads are necessary to work around python/mypy#3644.
 @overload
 def abspath(path: PathLike[AnyStr]) -> AnyStr: ...
 @overload
 def abspath(path: AnyStr) -> AnyStr: ...
 @overload
+def basename(p: LiteralString) -> LiteralString: ...
```

Comment:
Do we really need two overloads for functions like this?

```python3
@overload
def basename(p: LiteralString) -> LiteralString: ...
@overload
def basename(p: AnyStr) -> AnyStr: ...
```
This seems a bit ridiculous. If a function is typed with `(p: AnyStr) -> AnyStr`, doesn't it just take any string as an argument and return the same type?

Maybe a cleaner solution would be a new typevar that is just like `AnyStr`, but also can be a `LiteralString`. Modifying `AnyStr` doesn't seem to be a good idea, because functions like `abspath()` return something very different from what's passed in, and they are typed with `AnyStr`.

Reply:
Good point, we could add a new TypeVar ranging over str, bytes, and LiteralString to _typeshed.

Naming ideas: `AnyStrOrLiteral`, `AnyOrLiteralStr`, `AnyStrEx`